jwst/pathloss/tests/test_pathloss.py

```diff
@@ -0,0 +1,332 @@
+    # Where i is the iteration of np.arange(wavesize) which is the shape of the uniform
+    # data array.
+    comparison = np.array([1 +(float(i+1) - 1)*1 for i in np.arange(10)])
+    assert(np.all(wavelength==comparison))
+    
+    # The same array is returned in this case
+    assert(np.all(datmod.apertures[0].uniform_data == pathloss))
+
+
+def test_calculate_pathloss_vector_interpolation():
+    """Calculate the pathloss vector for when interpolation is necessary.
+    """
+    datmod = PathlossModel()
+
+    ref_data = {'pointsource_data':np.ones((10,10,10), dtype=np.float32),
+                'pointsource_wcs': {'crval2': -0.5, 'crpix2': 1.0, 'cdelt2': 0.5, 
+                                    'cdelt3': 1.0, 'crval1': -0.5, 'crpix1': 1.0, 
+                                    'crpix3': 1.0, 'crval3': 1.0, 'cdelt1': 0.5}}
+
+    datmod.apertures.append(ref_data)
+
+    wavelength, pathloss, is_inside_slitlet = calculate_pathloss_vector(datmod.apertures[0].pointsource_data,
+                                                                        datmod.apertures[0].pointsource_wcs,
+                                                                        0.0, 0.0)
+
+    # Wavelength array is calculated with this: crval3 +(float(i+1) - crpix3)*cdelt3
+    # Where i is the iteration of np.arange(wavesize) which is the 1st dimension of the pointsource
+    # data array.
+    wavelength_comparison = np.array([1 + (float(i+1) - 1.0)*1 for i in np.arange(10)])
+    assert(np.all(wavelength==wavelength_comparison))
+
+    # In this instance we interpolate to get the array for pathloss VS wavelength.
+    # With the current values inside of the of the pointsource_wcs starting at line 143 of pathloss.py
+    # dx1 = 1 - int(1) = 0.0
+    # dx2 = 1 - dx1 = 1.0
+    # dy1 = 1 - int(1) = 0.0
+    # dy2 = 1 - dx1 = 1.0
+    # This means a11 == a12 == a21 == 0 and a22 == 1
+    # This simplifies that pathloss vector to:
+    # pathloss_vector = (a22*pathloss_ref[:,i,j]) = (1*pathloss_ref[:1,j])
+    # Thus pathloss == the input array to the function.
+    pathloss_comparison = datmod.apertures[0].pointsource_data
+    assert(np.all(pathloss==pathloss_comparison))
+
+    # With the current wcs values, the logic should be returning True
+    assert(is_inside_slitlet==True)
+
+
+def test_is_pointsource():
+    """Check to see if object it point source
+    """
+    point_source = None
+    result = is_pointsource(point_source)
+    assert(result == False)
+
+    point_source = 'point'
+    result = is_pointsource(point_source)
+    assert(result == True)
+
+    point_source = 'not a point'
+    result = is_pointsource(point_source)
+    assert(result == False)
+
+# Begin do_correction exception testing
+def test_do_correction_msa_slit_size_eq_0():
+    """If slits have size 0, quit calibration.
+    """
+    datmod = MultiSlitModel()
+    datmod.slits.append({'data':np.empty((10,10,10))})
```

Comment:
`np.empty` doesn't give you an array with size 0; it returns an array with uninitialized data.

Reply:
@philhodge ahh right, I went ahead and replace the lines with `np.array([])`! Thanks